test/e2e/angular-app/prestart.js

```diff
@@ -0,0 +1,13 @@
+const path = require('path');
+const fs = require('fs-extra');
+
+const target = path.join(__dirname, 'src/environments/environment.ts');
+fs.ensureFileSync(target);
+fs.writeFileSync(target, `
+
+export const environment = {
+  production: true,
+  WIDGET_TEST_SERVER: '${process.env.WIDGET_TEST_SERVER}',
+};
+
+`);
```

Comment:
Is this needed?

Reply:
The `prestart.js` file is needed, because you can't pull environment variables from your angular js files. Anything in your `src` directory counts, so I have to create the environment file before I build.